client/js/lounge.js

```diff
@@ -444,9 +444,16 @@ $(function() {
 		}
 	});
 
-	sidebar.on("click", ".close", function() {
+	function closeChan(itemData) {
 		var cmd = "/close";
-		var chan = $(this).closest(".chan");
+		var chan;
+
+		if (typeof itemData === "undefined") {
+			chan = $(this).closest(".chan");
```

Comment:
In which situation can this be `undefined` actually?
(Also, we have a bunch of `=== undefined` in the code, I think you can go for that. I know, I know, what about if `undefined` gets reassigned, I've heard the urban legends :D)

Reply:
Whoops, looks like it won't be, you're right.

I've refactored it locally to simply pass the channel into the function:

```js
close: function(itemData) {
	closeChan($(`.networks .chan[data-target="${itemData}"]`));
},
```

I have one bug to track down before I commit, though, and I don't have time to do that right now, so it'll have to wait.

Reply:
If you want help with your bug, feel free to commit and describe it so I can reproduce + debug :)

Reply:
I figured it out 👍